### PR TITLE
VM mutating webhook fails on error

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
VM mutating webhook now fails on error.

**Release note**:
```release-note
None
```

### CODE REVIEW

**Comment by jean-edouard** [comment]
Code looks good. However if that webhook is not yet tested it probably should be!

File: pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go [hunk 1]
```diff
@@ -35,32 +36,22 @@ type VMsMutator struct {
...
-		Allowed: true,
-	}
-}
-
 func (mutator *VMsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
 	if !webhookutils.ValidateRequestResource(ar.Request.Resource, webhooks.VirtualMachineGroupVersionResource.Group, webhooks.VirtualMachineGroupVersionResource.Resource) {
-		log.Log.V(1).Warningf("vm-mutator: received invalid request")
-		return emptyValidResponse()
+		err := fmt.Errorf("expect resource to be '%s'", webhooks.VirtualMachineGroupVersionResource.Resource)
+		return webhookutils.ToAdmissionResponseError(err)
```
Comment (dankenigsberg) [inline pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go:42]: I don't know much about this webhook, but this change of semantics seems important and testable. If I'm correct, can you add a "negative" test for it?

Comment (dankenigsberg) [reply pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go:44]: thanks.

**Comment by akrejcir** [comment]
@jean-edouard can you take another look?

**Pull request merged**